ui/src/app/store/zone/types/base.ts

```diff
@@ -12,4 +15,41 @@ export type Zone = Model & {
   updated: string;
 };
 
-export type ZoneState = GenericState<Zone, APIError>;
+export type ZonePK = Zone[typeof ZONE_PK];
+
+export type ZoneProcess = {
+  processing: ZonePK[];
+  success: ZonePK[];
+};
+
+export type ZoneProcesses = {
+  [ZONE_ACTIONS.delete]: ZoneProcess;
+  [ZONE_ACTIONS.update]: ZoneProcess;
+};
+
+export type ZoneAPIMeta = {
+  formId: string | null;
+  modelPk: ZonePK | null;
+};
+
+export type ZoneAPISuccess<R> = {
+  result: R;
+};
+
+export type ZoneAPIError = ZoneAPIMeta & {
+  error: APIError;
+};
```

Comment:
These are all prefixed with `Zone` but the idea is for the standard websocket action shape to look like these.